changes/5796.changed

```diff
@@ -0,0 +1 @@
+Raise exception when trying to assign a non-child device to a device bay.
```

Comment:
I'd probably consider this a `.fixed` rather than a `.changed`, and word it something like:

```
Added missing validation to ensure that a DeviceBay can only contain Devices with a `subdevice_role` of `child`.
```